e2e/LongHaul/amqpPooling/IotHub.cs

```diff
@@ -0,0 +1,183 @@
+            _logger = logger;
+            _hubConnectionString = parameters.IotHubConnectionString;
+            _transportSettings = parameters.GetTransportSettingsWithPooling();
+            _clientOptions = new IotHubClientOptions(_transportSettings);
+
+            s_devices = devices;
+            s_deviceOperations = new Dictionary<string, DeviceOperations>();
+        }
+
+        public Dictionary<string, string> TelemetryUserProperties { get; } = new();
+
+        public async Task InitializeAsync()
+        {
+            await _lifetimeControl.WaitAsync().ConfigureAwait(false);
+
+            var helper = new IotHubConnectionStringHelper(_hubConnectionString);
+
+            try
+            {
+                _logger.Trace(
+                    $"Creating {s_devices.Count} device clients with transport settings [{_transportSettings.ToString()}].",
+                    TraceSeverity.Information);
+
+                foreach (var device in s_devices)
+                {
+                    string deviceConnectionString = $"HostName={helper.HostName};DeviceId={device.Id};SharedAccessKey={device.Authentication.SymmetricKey.PrimaryKey}";
+
+                    var deviceClient = new IotHubDeviceClient(deviceConnectionString, _clientOptions);
+
+                    // Set the device client connection status change handler
+                    var amqpConnectionStatusChange = new AmqpConnectionStatusChange(_logger.Clone());
+                    deviceClient.ConnectionStatusChangeCallback = amqpConnectionStatusChange.ConnectionStatusChangeHandler;
+
+                    await deviceClient.OpenAsync().ConfigureAwait(false);
+
+                    var deviceOperations = new DeviceOperations(deviceClient, device.Id, _logger.Clone());
+                    s_deviceOperations.Add(device.Id, deviceOperations);
+                }
+            }
+            finally
+            {
+                _lifetimeControl.Release();
+            }
+        }
+
+        public async Task SendTelemetryMessagesAsync(CancellationToken ct)
+        {
+            while (!ct.IsCancellationRequested)
+            {
+                await Task.Delay(s_messageLoopSleepTime, ct).ConfigureAwait(false);
+
+                List<Task> deviceOperationTasks = new();
+
+                _messagesToSend.TryDequeue(out SystemHealthTelemetry systemHealth);
+                if (systemHealth == null)
+                {
+                    continue;
+                }
+
+                if (s_deviceOperations != null)
+                {
+                    foreach (KeyValuePair<string, DeviceOperations> entry in s_deviceOperations)
+                    {
+                        string deviceId = entry.Key;
+                        DeviceOperations deviceOperations = entry.Value;
+
+                        var telemetryObject = new DeviceTelemetry
+                        {
+                            DeviceId = deviceId,
+                            SystemHealth = systemHealth,
```

Comment:
Probably not much reason for each device to publish system health, as they are all in the same process.

You could gather than regularly and log it in a loop as another task to kick off.

Reply:
With the comment below, I will just create and use another property within `DeviceTelemetry`, maybe something like "GuidValue".